numpy/_core/defchararray.py

```diff
@@ -2690,7 +840,7 @@ def partition(self, sep):
         """
         return asarray(partition(self, sep))
 
-    def replace(self, old, new, count=None):
+    def replace(self, old, new, count=-1):
```

Comment:
I think probably best to leave the default alone in a member function of `chararray` because it's effectively deprecated and is poorly tested. I think the `_check_args` function in `np._core.strings` handle dropping the `None` argument.

Reply:
Agreed but for a different reason: this PR should really only move code, not make any user-visible API changes. I think it *is* OK to do `replace(self, old, new, count if count is not None else -1)` below, though! 

p.s. One could also adjust `strings.replace` to allow `count=None` again (like it does in the function removed here), but I think that is less good, since the new definition, with its default of `-1`, matches the python `str.replace` default.